test/e2e/specs/plugins/plugins__browse-calypso-user.ts

```diff
@@ -74,6 +74,18 @@ describe( DataHelper.createSuiteTitle( 'Plugins: Browse' ), function () {
 		] )( 'Featured Plugins section should show the %s plugin', async function ( plugin: string ) {
 			await pluginsPage.validateHasPluginOnSection( 'featured', plugin );
 		} );
+
+		it( 'Can browse SEO category', async function () {
+			await pluginsPage.clickCategoryButton( 'Search Engine Optimization' );
+			await pluginsPage.validateHasSubtitle( 'Search Engine Optimization' );
```

Comment:
**suggestion**: it would be cleaner and possibly less error prone if the validation occurred within the main click action. 

Something like this will remove the need for validation methods:

```
const locator = page.locator( subtitle )
await locator.waitFor()
```

Reply:
I've seen similar suggestions from @dpasque before, for e2e tests is there a sort of general best practice to make assertions in the same place as we make the actions (in page components) leaving our test specs mostly assertion free and more like test scripts than like a unit test where you make an action and then an assertion to test the action completed? This then has the benefit of not having to think about making assertions when you make repeated actions in other specs ("scripts").

(Just wondering if that's the right way I should explain / think about it)

Reply:
Thanks, @worldomonation, I made it and also fixed mobile e2e tests

Reply:
> I've seen similar suggestions from @dpasque before, for e2e tests is there a sort of general best practice to make assertions in the same place as we make the actions (in page components) leaving our test specs mostly assertion free and more like test scripts than like a unit test where you make an action and then an assertion to test the action completed? This then has the benefit of not having to think about making assertions when you make repeated actions in other specs ("scripts").
> 
> 
> 
> (Just wondering if that's the right way I should explain / think about it)

Correct, it is preferable for the E2E tests to read like a series of actions that are expected to happen, kind of like "let this event occur" 

eg.
- land on checkout
- make purchase 
- see congratulations

Reply:
> Correct, it is preferable for the E2E tests to read like a series of actions that are expected to happen, kind of like "let this event occur"

eg.

land on checkout
make purchase
see congratulations

That is really good advice thanks @worldomonation